<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from kuka_arm.xacro                 | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="iis_robot" xmlns:xacro="http://ros.org/wiki/xacro">
  <material name="DarkGrey">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <material name="LightGrey">
    <color rgba="0.8 0.8 0.8 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <material name="Black">
    <color rgba="0 0 0 1.0"/>
  </material>
  <material name="Orange">
    <color rgba="1.0 0.487 0 1.0"/>
  </material>
  <material name="Silver">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="HandGray">
    <color rgba="0.953 0.996 0.694 1.0"/>
  </material>
  <material name="SickBlue">
    <color rgba="0.3058 0.5921 0.7294 1.0"/>
  </material>
  <!-- draw a small cross to visualize the origin of the world reference frame -->
  <link name="world_link"/>
  <!-- arm with fake eef -->
  <joint name="world_kuka_arm_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world_link"/>
    <child link="calib_kuka_arm_base_link"/>
  </joint>
  <link name="calib_kuka_arm_base_link"/>
  <joint name="calib_kuka_arm_base_joint" type="fixed">
    <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
    <parent link="calib_kuka_arm_base_link"/>
    <child link="kuka_arm_base_link"/>
  </joint>
  <link name="kuka_arm_base_link">
    <inertial>
      <mass value="2.0"/>
      <origin xyz="0 0 0.055"/>
      <inertia ixx="0.00381666666667" ixy="0" ixz="0" iyy="0.0036" iyz="0" izz="0.00381666666667"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/base.STL"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0.055"/>
      <geometry>
        <!--          <mesh filename="package://lwr_description/meshes/lwr/convex/arm_base_convex.stl"/> -->
        <cylinder length="0.11" radius="0.07"/>
      </geometry>
    </collision>
  </link>
  <joint name="kuka_arm_0_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.11"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="0.191986217719"/>
    <safety_controller k_position="30" k_velocity="30" soft_lower_limit="-2.93215314335" soft_upper_limit="2.93215314335"/>
    <dynamics damping="1.0"/>
    <parent link="kuka_arm_base_link"/>
    <child link="kuka_arm_1_link"/>
  </joint>
  <link name="kuka_arm_1_link">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 -0.06 0.130"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.003" iyz="0" izz="0.0118666666667"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.008"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/link_1.STL"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <!--
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://lwr_description/meshes/lwr/convex/arm_segment_a_convex.stl"/>
        </geometry>
      </collision>
 -->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.06 0.2"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.291 0 0" xyz="0 -0.03 0.1"/>
      <geometry>
        <cylinder length="0.208" radius="0.07"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="kuka_arm_1_link">
    <gravity>false</gravity>
  </gazebo>
  <transmission name="kuka_arm_0_trans" type="SimpleTransmission">
    <actuator name="kuka_arm_0_motor"/>
    <joint name="kuka_arm_0_joint"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="kuka_arm_1_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.2005"/>
    <axis xyz="0 1 0"/>
    <limit effort="306" lower="-2.09439510239" upper="2.09439510239" velocity="0.191986217719"/>
    <safety_controller k_position="30" k_velocity="30" soft_lower_limit="-2.05948851735" soft_upper_limit="2.05948851735"/>
    <dynamics damping="1.0" friction="10"/>
    <parent link="kuka_arm_1_link"/>
    <child link="kuka_arm_2_link"/>
  </joint>
  <link name="kuka_arm_2_link">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 0.06 0.07"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.003" iyz="0" izz="0.0118666666667"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/link_2.STL"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <!--
      <collision>
        <origin xyz="0 0 0.2" rpy="${M_PI} 0 0"/>
        <geometry>
          <mesh filename="package://lwr_description/meshes/lwr/convex/arm_segment_b_convex.stl"/>
        </geometry>
      </collision>
-->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.06 0.0"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.291 0 0" xyz="0 0.03 0.1"/>
      <geometry>
        <cylinder length="0.208" radius="0.07"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="kuka_arm_2_link">
    <gravity>false</gravity>
  </gazebo>
  <transmission name="kuka_arm_1_trans" type="SimpleTransmission">
    <actuator name="kuka_arm_1_motor"/>
    <joint name="kuka_arm_1_joint"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="kuka_arm_2_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="0.226892802759"/>
    <safety_controller k_position="30" k_velocity="30" soft_lower_limit="-2.93215314335" soft_upper_limit="2.93215314335"/>
    <dynamics damping="1.0"/>
    <parent link="kuka_arm_2_link"/>
    <child link="kuka_arm_3_link"/>
  </joint>
  <link name="kuka_arm_3_link">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 0.06 0.130"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.003" iyz="0" izz="0.0118666666667"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.008"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/link_3.STL"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <!--
      <collision>
        <origin xyz="0 0 0" rpy="0 0 ${M_PI}"/>
        <geometry>
          <mesh filename="package://lwr_description/meshes/lwr/convex/arm_segment_a_convex.stl"/>
        </geometry>
      </collision>
-->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.06 0.2"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="-0.291 0 0" xyz="0 0.03 0.1"/>
      <geometry>
        <cylinder length="0.208" radius="0.07"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="kuka_arm_3_link">
    <gravity>false</gravity>
  </gazebo>
  <transmission name="kuka_arm_2_trans" type="SimpleTransmission">
    <actuator name="kuka_arm_2_motor"/>
    <joint name="kuka_arm_2_joint"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="kuka_arm_3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20"/>
    <axis xyz="0 -1 0"/>
    <limit effort="306" lower="-2.09439510239" upper="2.09439510239" velocity="0.226892802759"/>
    <safety_controller k_position="30" k_velocity="30" soft_lower_limit="-2.05948851735" soft_upper_limit="2.05948851735"/>
    <dynamics damping="1.0"/>
    <parent link="kuka_arm_3_link"/>
    <child link="kuka_arm_4_link"/>
  </joint>
  <link name="kuka_arm_4_link">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 -0.06 0.07"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.003" iyz="0" izz="0.0118666666667"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/link_4.STL"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <!--
      <collision>
        <origin xyz="0 0 0.2" rpy="0 ${M_PI} 0"/>
        <geometry>
          <mesh filename="package://lwr_description/meshes/lwr/convex/arm_segment_b_convex.stl"/>
        </geometry>
      </collision>
-->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.06 0.0"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="-0.291 0 0" xyz="0 -0.03 0.1"/>
      <geometry>
        <cylinder length="0.208" radius="0.07"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="kuka_arm_4_link">
    <gravity>false</gravity>
  </gazebo>
  <transmission name="kuka_arm_3_trans" type="SimpleTransmission">
    <actuator name="kuka_arm_3_motor"/>
    <joint name="kuka_arm_3_joint"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="kuka_arm_4_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="0.226892802759"/>
    <safety_controller k_position="30" k_velocity="30" soft_lower_limit="-2.93215314335" soft_upper_limit="2.93215314335"/>
    <dynamics damping="1.0"/>
    <parent link="kuka_arm_4_link"/>
    <child link="kuka_arm_5_link"/>
  </joint>
  <link name="kuka_arm_5_link">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 0 0.124"/>
      <inertia ixx="0.0126506666667" ixy="0" ixz="0" iyy="0.003" iyz="0" izz="0.0108506666667"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.008"/>
      <geometry name="kuka_arm_5_geom">
        <mesh filename="package://lwr_description/meshes/lwr/link_5.STL"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <!--
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://lwr_description/meshes/lwr/convex/arm_segment_last_convex.stl"/>
        </geometry>
      </collision>
-->
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.04 0.19"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.207 0 0" xyz="0 -0.02 0.095"/>
      <geometry>
        <cylinder length="0.198" radius="0.07"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="kuka_arm_5_link">
    <gravity>false</gravity>
  </gazebo>
  <transmission name="kuka_arm_4_trans" type="SimpleTransmission">
    <actuator name="kuka_arm_4_motor"/>
    <joint name="kuka_arm_4_joint"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="kuka_arm_5_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.19"/>
    <axis xyz="0 1 0"/>
    <limit effort="306" lower="-2.09439510239" upper="2.09439510239" velocity="0.314159265359"/>
    <safety_controller k_position="30" k_velocity="30" soft_lower_limit="-2.05948851735" soft_upper_limit="2.05948851735"/>
    <dynamics damping="1.0"/>
    <parent link="kuka_arm_5_link"/>
    <child link="kuka_arm_6_link"/>
  </joint>
  <link name="kuka_arm_6_link">
    <inertial>
      <mass value="0.2"/>
      <origin rpy="0 0 0" xyz="0 0 0.0625"/>
      <inertia ixx="0.000520833333333" ixy="0" ixz="0" iyy="0.000520833333333" iyz="0" izz="0.000520833333333"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/link_6.STL"/>
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/convex/arm_wrist_convex.stl"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="kuka_arm_6_link">
    <gravity>false</gravity>
  </gazebo>
  <transmission name="kuka_arm_5_trans" type="SimpleTransmission">
    <actuator name="kuka_arm_5_motor"/>
    <joint name="kuka_arm_5_joint"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="kuka_arm_6_joint" type="revolute">
    <origin rpy="0 0 3.14159265359" xyz="0 0 0.078"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="0.314159265359"/>
    <safety_controller k_position="30" k_velocity="30" soft_lower_limit="-2.93215314335" soft_upper_limit="2.93215314335"/>
    <dynamics damping="1.0"/>
    <parent link="kuka_arm_6_link"/>
    <child link="kuka_arm_7_link"/>
  </joint>
  <link name="kuka_arm_7_link">
    <inertial>
      <mass value="0.2"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0333333333333" ixy="0" ixz="0" iyy="0.0333333333333" iyz="0" izz="0.0333333333333"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://lwr_description/meshes/lwr/link_7.STL"/>
      </geometry>
      <material name="DarkGrey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0.785398163397" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.04"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="kuka_arm_7_link">
    <gravity>false</gravity>
  </gazebo>
  <transmission name="kuka_arm_6_trans" type="SimpleTransmission">
    <actuator name="kuka_arm_6_motor"/>
    <joint name="kuka_arm_6_joint"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="eef_joint" type="fixed">
    <origin rpy="0 0 0"/>
    <parent link="kuka_arm_7_link"/>
    <child link="eef"/>
  </joint>
  <link name="eef"/>
</robot>

